src/main/java/com/minecolonies/coremod/colony/requestsystem/locations/StaticLocation.java

```diff
@@ -0,0 +1,154 @@
+package com.minecolonies.coremod.colony.requestsystem.locations;
+
+import com.google.common.reflect.TypeToken;
+import com.minecolonies.api.colony.requestsystem.factory.IFactoryController;
+import com.minecolonies.api.colony.requestsystem.location.ILocation;
+import com.minecolonies.api.colony.requestsystem.location.ILocationFactory;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.math.BlockPos;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Location described by an immutable blockpos and a dimension.
+ */
+public class StaticLocation implements ILocation
+{
+
+    @NotNull
+    private final BlockPos pos;
+
+    private final int dimension;
+
+    StaticLocation(@NotNull final BlockPos pos, final int dimension)
+    {
+        this.pos = pos;
+        this.dimension = dimension;
+    }
+
+    /**
+     * Method to get the location in the dimension
+     *
+     * @return The location.
+     */
+    @NotNull
+    @Override
+    public BlockPos getInDimensionLocation()
+    {
+        return pos;
+    }
+
+    /**
+     * Method to get the dimension of the location.
+     *
+     * @return The dimension of the location.
+     */
+    @Override
+    public int getDimension()
+    {
+        return dimension;
+    }
+
+    /**
+     * Method to check if this location is reachable from the other.
+     *
+     * @param location The check if it is reachable from here.
+     * @return True when reachable, false when not.
+     */
+    @Override
+    public boolean isReachableFromLocation(@NotNull final ILocation location)
+    {
+        return location.getDimension() == getDimension();
+    }
+
+    /**
+     * Internal factory class.
+     */
+    public static class Factory implements ILocationFactory<BlockPos, StaticLocation>
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 54 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)